File: src/javaexamples/Person.java
package javaexamples;

public class Person {
  private final int id;
  private String name;

  public Person(int id, String name) {
    this.id = id;
    setName(name);
    }

  public int getId() { return id; }

  public String getName() { return name; }

  public void setName(String value) {
    if (value == null) throw new RuntimeException();
    name = value;
    }

  }

File: src/Person.kt
class Person(id: Int, name: String) {
  val id = id

  var name = name

  }

fun foo1(a: Int, b: Int?): Int {
  if (b == null) return 0
  return a*b
  }

fun foo2(a: Int, b: Int?) =
  if (b == null) 0
  else a*b

